polar_route/crossing.py

```diff
@@ -85,34 +85,35 @@ def __init__(self,source_graph=None,neighbour_graph=None,
         self.debugging     = debugging
 
     def _newton_optimisation(self, f, x, a, Y, u1, v1, u2, v2, s1, s2):
-        """
-            Find the y value of the crossing point and travel time in each cell using the Newton method
-        """
+        '''
+            FILL
+        '''
         y0 = (Y*x)/(x+a)
         if self.debugging:
             print('---Initial y={:.2f}'.format(y0))
         improving = True
-        iteration_num = 0
+        iterartion_num = 0
         while improving:
-            F, dF, X1, X2, t1, t2  = f(y0,x,a,Y,u1,v1,u2,v2,s1,s2)
-            if F == np.inf:
-                return np.nan, np.inf
+            F,dF,X1,X2,t1,t2  = f(y0,x,a,Y,u1,v1,u2,v2,s1,s2)
+            if F==np.inf:
+                print('F inf')
+                return np.nan,np.inf  
 
             if self.debugging:
                 print('---Iteration {}: y={:.2f}; F={:.5f}; dF={:.2f}'\
-                      .format(iteration_num, y0, F, dF))
+                      .format(iterartion_num,y0,F,dF))
             y0  = y0 - (F/dF)
             improving = abs((F/dF)/(X1*X2)) > self.optimizer_tol
-            iteration_num += 1
-            # Assume convergence impossible after 1000 iterations and exit
-            if iteration_num > 1000:
-                # Set crossing point to nan and travel times to infinity
```

Comment:
It would be better to update the comments I wrote here if they're out of date, rather than just deleting them

Reply:
Comments have been re-included